homeassistant/components/mazda/config_flow.py

```diff
@@ -0,0 +1,123 @@
+                    user_input[CONF_PASSWORD],
+                    user_input[CONF_REGION],
+                    websession,
+                )
+                await mazda_client.validate_credentials()
+            except MazdaAuthenticationException:
+                errors["base"] = "invalid_auth"
+            except MazdaAccountLockedException:
+                errors["base"] = "account_locked"
+            except aiohttp.ClientError:
+                errors["base"] = "cannot_connect"
+            except Exception as ex:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+                _LOGGER.exception(
+                    "Unknown error occurred during Mazda login request: %s", ex
+                )
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_EMAIL], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth if the user credentials have changed."""
+        errors = {}
+
+        if (
+            user_input is not None
+            and user_input.get(CONF_EMAIL) is not None
+            and user_input.get(CONF_PASSWORD) is not None
+            and user_input.get(CONF_REGION) is not None
+        ):
+            try:
+                websession = aiohttp_client.async_get_clientsession(self.hass)
+                mazda_client = MazdaAPI(
+                    user_input[CONF_EMAIL],
+                    user_input[CONF_PASSWORD],
+                    user_input[CONF_REGION],
+                    websession,
+                )
+                await mazda_client.validate_credentials()
+            except MazdaAuthenticationException:
+                errors["base"] = "invalid_auth"
+            except MazdaAccountLockedException:
+                errors["base"] = "account_locked"
+            except aiohttp.ClientError:
+                errors["base"] = "cannot_connect"
+            except Exception as ex:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+                _LOGGER.exception(
+                    "Unknown error occurred during Mazda login request: %s", ex
+                )
+            else:
+                await self.async_set_unique_id(user_input[CONF_EMAIL].lower())
+
+                for entry in self._async_current_entries():
+                    if entry.unique_id == self.unique_id:
+                        self.hass.config_entries.async_update_entry(
+                            entry, data=user_input
+                        )
+
+                        # Reload the config entry otherwise devices will remain unavailable
+                        self.hass.async_create_task(
+                            self.hass.config_entries.async_reload(entry.entry_id)
+                        )
```

Comment:
You've re-implemented the exact logic of [`self._abort_if_unique_id_configured`](https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L898) so use that method instead.

Reply:
It looks like `self._abort_if_unique_id_configured` aborts with reason `"already_configured"`. But I am doing reauth, so I need to abort with reason `"reauth_successful"` in order to display the correct message to the user, right? Unless I'm missing something here.

Reply:
You're right. My bad.